java/piranha/src/main/java/com/uber/piranha/XPFlagCleaner.java

```diff
@@ -286,7 +286,7 @@ private API getXPAPI(ExpressionTree et) {
         return API.UNKNOWN;
       }
 
-      if (mit.getArguments().size() == 1 || mit.getArguments().size() == 2) {
+      if (mit.getArguments().size() >= 1) {
```

Comment:
Let us switch the search for arguments and methods i.e., search for the method names first and if they match, check for flag name. Otherwise, I am afraid this operation will become quite expensive. 

Also, better to move all the method HashSets (`controlMethods`, `treatedMethods`, etc) into a HashMap<String,API>.